DeAceroDLabs/marketplace-frontend
# [DMP-612] Get SKU from selected properties

## What is done

[[DMP-612] Obtener SKU desde propiedades seleccionadas](https://deacero.atlassian.net/browse/DMP-612)

## How is done

1. Redefined some types in the api.ts file to match with what is returned from the service.
2. Added some functions to handle the selection of a property value.
  - A useEffect is used to get the SKU from the selected properties from the user. This filters all the articles by the last selected property and then within these check for an article that matches all selected properties 
  - A useEffect is used to only get the product details from the service and its characteristics.
  - A function `isPropertyAlreadyIncluded` is used to check if the user has already selected a property value.
  - A function `selectProperty` first checks if a property value has already been selected by the user, if not then adds this to the selected properties array by default, otherwise substitutes the value of the property within the old property.
3. Added a new section in the view to display the selected values from the user, if a matching SKU exists, this is displayed at the top of the properties. Otherwise, it displays a "No existe producto" message

## How to test

1. Run the project via `npm start`
2. Select a product from the "catálogo de productos" section, I recommend selecting the first "clavo".
3. Wait for the details to load into the view. 
4. Open each property section and select a value. 
5. Check how the value is added to its corresponding field in the "Propiedades del Producto" section.
6. When an existing article is found by selecting certain properties values, the SKU will appear.

Here are some examples of products with their SKU and valid combination of values

```
Clavo SKU 3586

Reforzado
Calibre 11
Largo 51
Presentation 25kg

Clavo SKU 3122

Con cabeza
Calibre 8
Largo 89
Presentation 25kg
```

**NOTE:** An important change that needs to be done is to refactor this view as the component has grown a lot. But that would be worked in a separate ticket.
